src/ext/java/org/opentripplanner/ext/emissions/EmissionsFilter.java

```diff
@@ -43,20 +40,17 @@ public List<Itinerary> filter(List<Itinerary> itineraries) {
         .filter(leg -> leg.getMode() == TraverseMode.CAR)
         .toList();
 
-      calculateCo2EmissionsForCar(carLegs)
-        .ifPresent(co2 -> {
-          if (transitLegs.isEmpty()) {
-            itinerary.setEmissionsPerPerson(new Emissions(co2));
-          } else if (
-            itinerary.getEmissionsPerPerson() != null &&
-            itinerary.getEmissionsPerPerson().getCo2() != null
-          ) {
-            // Calculate park and ride emissions.
-            itinerary.setEmissionsPerPerson(
-              new Emissions(itinerary.getEmissionsPerPerson().getCo2().plus(co2))
-            );
-          }
-        });
+      Optional<Grams> co2ForCar = calculateCo2EmissionsForCar(carLegs);
+
+      if (
+        co2ForTransit.isPresent() ||
+        co2ForCar.isPresent() &&
+        (transitLegs.isEmpty() || co2ForTransit.isPresent())
```

Comment:
I think it could improve readibility if this was wrapped with `()` so you don't have to really pay that much attention how this and the `||` before this combine. Maybe it could even make sense to split this if into two different cases, one where there is no transit and car emissions are available and one where there are both. Then you don't have to do the if present checks again inside this.

Reply:
And you check `co2ForTransit` several times. You should keep an eye out for yellow lines in IntelliJ.

![image](https://github.com/opentripplanner/OpenTripPlanner/assets/151346/7a6e42d5-3289-470b-b521-982b0b06e1aa)